### PR TITLE
make discourse logger config a templated secret

### PR DESCRIPTION
**Pull request by cmharlow**
Jira: https://mozilla-hub.atlassian.net/browse/SE-2173 

What this PR does:
* replaces the hard-coded logger config ConfigMap template with an ExternalSecrets setup (update discourse helm chart, add secret metadata to terraform for discourse prod logger secret)
* came up in decommissioning https://github.com/mozilla-it/discourse-infra

Related to: https://github.com/mozilla-it/helm-charts/pull/121 & https://github.com/mozilla-it/itse-apps-stage-1-infra/pull/107

### CODE REVIEW

**Review by floatingatoll — APPROVED** [review]

**Comment by cmharlow** [comment]
waiting on successful deploys in dev, stage

**Review by floatingatoll — APPROVED** [review]

**Comment by cmharlow** [comment]
terraform successfully applied. uncovered old cdn origin still in terraform for prod, removed (had already been removed manually after prod migration to avoid dangling CNAMEs)

**Comment by cmharlow** [comment]
Waiting until evening Eastern time to merge & finalize. Deploy update shouldn't affect users, but see we have 30 or so users on the site at this moment, so being cautious.

**Pull request merged**
